front: rename Occurrence.exceptions to exceptionChangeGroups

emersion opened the pull request:
_See individual commits._

SharglutDev commented:
Debatable :) Maybe I should have rather kept the term `exceptionChangeGroups` for the property name

emersion commented:
In that case `OccurrenceException` is not a good name?

Either way, I find the status quo pretty confusing.

SharglutDev commented:
Yep `OccurrenceException` is also confusing :x Could have just been `ExceptionChangeGroups`

emersion commented:
Okay, so I'll rename:

- `ExceptionChangeGroup` to `ExceptionChangeGroupName`
- `OccurrenceExceptionChangeGroups` can be merged together with `ExceptionChangeGroup` I believe (it's the same union)
- `Occurrence.exceptions` to `exceptionChangeGroups`
- `OccurrenceException` to `ExceptionChangeGroups`

SharglutDev commented:
> Okay, so I'll rename:
> 
>    `OccurrenceExceptionChangeGroups` can be merged together with `ExceptionChangeGroup` I believe (it's the same union)

<del>Hmm one corresponds to the editoast format and one to the translation key format</del>
Misread, we can do that yes :)

emersion committed:
front: simplify OccurrenceExceptionChangeGroups

No need for Extract here: all keys are strings.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: rename OccurrenceExceptionChangeGroups to ExceptionChangeGroupName

Make it clear this is not holding actual change group values, just
the key names.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop ExceptionChangeGroup

I don't think it's worth maintaining a separate list and mapping
table just to have slightly more aesthetic names.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: rename OccurrenceException to ExceptionChangeGroups

This is more explicit because this type doesn't hold the full
exception, only change groups.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: rename Occurrence.exceptions to exceptionChangeGroups

This object represents a single exception, and only contains the
exception's change groups (not the full exception).

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
Done!

SharglutDev reviewed (approved):
Lgtm and tested :white_check_mark:
